addons/test_website/tests/test_redirect.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.tests import HttpCase, tagged
+
+
+@tagged('-at_install', 'post_install')
+class TestRedirect(HttpCase):
+
+    def setUp(self):
+        super(TestRedirect, self).setUp()
+
+        self.user_portal = self.env['res.users'].with_context({'no_reset_password': True}).create({
+            'name': 'Test Website Portal User',
+            'login': 'portal_user',
+            'password': 'portal_user',
+            'email': 'portal_user@mail.com',
+            'groups_id': [(6, 0, [self.env.ref('base.group_portal').id])]
+        })
+        self.redirect = self.env['website.rewrite'].create({
+            'name': 'Test Website Redirect',
+            'redirect_type': '308',
+            'url_from': '/test_website/country/<model("res.country"):country>',
+            'url_to': '/redirected/country/<model("res.country"):country>',
+        })
+
+    def test_01_redirect_308_model_converter(self):
+        """ Ensure 308 redirect with model converter works fine, including:
+                - Correct & working redirect as public user
+                - Correct & working redirect as logged in user
+                - Correct replace of url_for() URLs in DOM
+        """
+        url = '/test_website/country/andorra-1'
```

Comment:
Andorra == id 1 is a little short.
Maybe search the first country e.g. ? or create one ...?

Reply:
Indeed I was a bit too lazy here